vpr/src/route/rr_graph2.cpp

```diff
@@ -1098,17 +1151,23 @@ static void load_block_rr_indices(const DeviceGrid& grid,
                                         indices[IPIN][x_tile][y_tile][side].push_back(OPEN);
                                     } else {
                                         VTR_ASSERT(class_type == RECEIVER);
-                                        indices[IPIN][x_tile][y_tile][side].push_back(*index);
                                         indices[OPIN][x_tile][y_tile][side].push_back(OPEN);
+                                        indices[IPIN][x_tile][y_tile][side].push_back(*index);
                                     }
-                                    ++(*index);
                                 } else {
                                     indices[IPIN][x_tile][y_tile][side].push_back(OPEN);
                                     indices[OPIN][x_tile][y_tile][side].push_back(OPEN);
                                 }
                             }
                         }
                     }
+                    /* A pin may locate on multiple sides of a tile.
+                     * Instead of allocating multiple rr_nodes for the pin,
+                     * we just create a rr_node and make it indexable on these sides
+                     * As such, we can avoid redundant rr_node to be allocated
+                     * and multiple nets to be mapped to the pin
+                     */
+                    ++(*index);
```

Comment:
Could be useful to have a check that we did in fact put this index in at least one spot (have a flag and assert if it is still false).